packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts

```diff
@@ -143,114 +241,51 @@ export async function handleAntiRaidNuke(
 				attachment: Buffer.from(formatMembersToAttachment(members, locale)),
 			},
 		],
-		components: [createMessageActionRow([cancelButton, banButton, dryRunButton])],
+		components: [row],
 	});
 
-	const collectedInteraction = (await reply
-		.awaitMessageComponent({
+	const buttonCollector = reply
+		.createMessageComponentCollector({
 			filter: (collected) => collected.user.id === interaction.user.id,
 			componentType: ComponentType.Button,
-			time: 60000,
+			time: ANTI_RAID_NUKE_COLLECTOR_TIMEOUT_SECONDS * 1000,
 		})
-		.catch(async () => {
+		.on('collect', async (collectedInteraction: ButtonInteraction<'cached'>) => {
+			switch (collectedInteraction.customId) {
+				case cancelKey: {
+					await collectedInteraction.update({
+						content: i18next.t('command.mod.anti_raid_nuke.common.cancel', {
+							lng: locale,
+						}),
+						components: [],
+					});
+					buttonCollector.stop('cancel');
+					break;
+				}
+
+				case banKey: {
+					await launchNuke(collectedInteraction, members, pruneDays, insensitive, mode, args, locale);
+					buttonCollector.stop('nuke');
+					break;
+				}
+				case dryRunKey: {
+					await sendDryRunResult(collectedInteraction, members, pruneDays, insensitive, mode, args, locale);
```

Comment:
Why the dry run does not stop the collector?

Reply:
I actually think that it should stop the collector if ephemeral

Reply:
More and more leaning towards running dry-run before even presenting ban and cancel, and just not presenting the buttons at all, if hidden, effectively replacing the preview with a dry-run

Reply:
I actually think this is a big brain ideia